src/App/Components/like.accommodation.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faHeart, faBookmark } from '@fortawesome/free-solid-svg-icons';
+import Apis from '../../Api';
+import checkToken, { token } from '../../helper/helper';
+
+class LikeAndBookMark extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			getLikeStatus: '0',
+			likeStatus: '0',
+		};
+		this.getLikes = this.getLikes.bind(this);
+		this.like = this.like.bind(this);
+	}
+	async getLikes() {
+		const { id } = this.props;
+		try {
+			const res = await Apis.getLikesOfAccommodation(id, token);
+			const { userId } = checkToken();
+			const liked = (res.data.data || []).map(element => element.userId).indexOf(userId);
+			if (liked !== -1) {
+				this.setState({
+					likeStatus: '1',
+				});
+			} else {
+				this.setState({
+					likeStatus: '0',
+				});
+			}
+			this.setState({
+				getLikeStatus: (res.data.data || []).length,
+			});
+		} catch (error) {}
+	}
+	async like() {
+		const { id } = this.props;
+		try {
+			await Apis.likeAccommodation(id, token);
+			this.setState({
+				likeStatus: '1',
+			});
+			this.getLikes();
+		} catch (err) {}
```

Comment:
I suggest to toast an error message in case there is any. to let the user know what's happening

Reply:
That was a nice idea;
But I don't think it's necessary.. cases like `Facebook` or `Instagram`.
The users always observe it, through changing Icon and number of likes